engine/src/main/java/com/datatorrent/stram/client/DTConfiguration.java

```diff
@@ -275,36 +275,32 @@ public void remove(String key)
     map.remove(key);
   }
 
-  public ValueEntry setInternal(String key, String value)
+  public void setInternal(String key, String value)
   {
-    ValueEntry valueEntry;
-    if (map.containsKey(key)) {
-      valueEntry = map.get(key);
+    ValueEntry valueEntry = map.get(key);
+    if (valueEntry != null) {
       valueEntry.value = value;
     } else {
       valueEntry = new ValueEntry();
```

Comment:
@vrozov I assume you meant when key is absent (else clause). Added the unit test.